### PR TITLE
Upgrade to angular 16

### PR DESCRIPTION
**Pull request by mpaczesny**
Upgrade to Angular 16

### CODE REVIEW

**Comment by Quantas** [comment]
I fixed the sample site, tests, coverage, devcontainer and the build (switched to GitHub actions to keep it all in one place).

@mpaczesny - are we sure we don't want any peerDeps any more? and just let me know if my changes look good and we can merge and release 7.0.0.

Thanks!!

**Comment by Quantas** [comment]
Also, I'm not too concerned about the drop in the code coverage at this time, unless you want to write more tests, the results are here: https://coveralls.io/builds/67149388

**Comment by Quantas** [comment]
Also, also, I published 7.0.0-SNAPSHOT for your testing :)

**Comment by Quantas** [comment]
Okay, there was a mixup, the current version published for testing is `7.0.0-rc1`

**Comment by mpaczesny** [comment]
I'll get the peer dependencies put back in. I think the ng update may have clobbered that.

**Comment by Quantas** [comment]
I'm going to add a commit and release 7.0.0-rc2, if you want to test installing/running that. If it looks good, I can merge this PR and release 7.0.0. Thanks @mpaczesny !

File: package.json
Comment (mpaczesny) [inline package.json:1]: @Quantas I'm running into an issue with the css files that are packaged with the code. 

Receiving this error: Package path ./shark-ng-table.min.css is not exported from package

Wondering if we need to do an exports section in the package.json like seen here: https://github.com/DepickereSven/ng-zorro-antd/commit/e91cf17105e6d8f0cfd6b023ee76aef275eacae4

Have you ever run into anything like this before?

**Comment by Quantas** [comment]
@mpaczesny  I published rc3, I did see that the .css files existed in rc2 as well, let me know how it goes.

**Comment by mpaczesny** [comment]
@Quantas I've completed my testing and everything looks good to me. Will you be able to cut the next release version sometime soon? We are hoping to implement tonight if possible. Sorry for the short notice.

**Comment by Quantas** [comment]
@mpaczesny this has been completed :)

**Pull request merged**